### PR TITLE
Implement Reclaim command for Legacy module - Closes #588

### LINKED ISSUE #588: Implement Reclaim command for Legacy module
### Description

Implement Reclaim command class under commands directory `reclaim.ts` and methods as [specified](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0050.md#reclaim)

### Acceptance Criteria

- All functions are implemented as per specification
- All functions are fully unit tested

### Additional Information
- Ensure the following scenarios are covered:

reclaim command | Expectation | Comments
-- | -- | --
When a user send reclaim transaction (with exact amount) from an address corresponding to an entry in the legacy accounts substore. | The mint function from the token module is called with the correct values. The entry in the legacy accounts substore is removed. |  
When a user send reclaim transaction (with invalid amount) from an address corresponding to an entry in the legacy accounts substore. | The transaction should be rejected with invalid amount claim error |  
When a user send reclaim transaction from an address corresponding to NO entry in the legacy accounts substore. | The transaction should be rejected with legacy account not found error |  
Params property does not follow the reclaimParamsSchema | Transaction rejected | Not sure if it should be part of the module tests, or be part of the base transactions tests

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #588 

### How was it solved?
- [x] Add implementation for `ReclaimCommand`
- [x] Add unit test cases for all necessary scenarios

### How was it tested?
- [x] Local
- [x] Jenkins

### CODE REVIEW

File: src/application/modules/legacy/commands/reclaim.ts [hunk 1]
```diff
@@ -12,13 +12,83 @@
...
+			);
+
+		await legacyStore.del(legacyAddress);
 
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	public async execute(): Promise<void> {}
+		await this._tokenAPI.mint(
+			ctx.getAPIContext(),
+			getAddressFromPublicKey(ctx.transaction.senderPublicKey),
+			{ chainID: CHAIN_ID, localID: LOCAL_ID },
```
Comment (ManuGowda) [inline src/application/modules/legacy/commands/reclaim.ts:90]: you need to access this after initialising.
1. in legacy module you should init like this https://github.com/LiskHQ/lisk-sdk/blob/3b5c7507f07c97754efd048c45c3b64ca4ddb99c/framework/src/modules/dpos_v2/module.ts#L135
2. in command u should like this https://github.com/LiskHQ/lisk-sdk/blob/3b5c7507f07c97754efd048c45c3b64ca4ddb99c/framework/src/modules/dpos_v2/commands/pom.ts#L59
3. then access like https://github.com/LiskHQ/lisk-sdk/blob/3b5c7507f07c97754efd048c45c3b64ca4ddb99c/framework/src/modules/dpos_v2/commands/pom.ts#L192


if you need help maybe check with one of SDK team members, to better understand the reasoning. 🙏🏼

**Review by ManuGowda — CHANGES_REQUESTED** [review]
@nagdahimanshu the code needs to be changed, please check with me or sdk team members for help

File: package-lock.json [hunk 2]
```diff
@@ -4,91 +4,77 @@
 	"lockfileVersion": 1,
 	"requires": true,
 	"dependencies": {
+		"@ampproject/remapping": {
```
Comment (ManuGowda) [inline package-lock.json]: why is the package-lock changed 🤔 ?

**Review by ManuGowda — CHANGES_REQUESTED** [review]
code change LGTM, good job @nagdahimanshu 👍🏼

File: package-lock.json (hunk 2 above)
Comment (nagdahimanshu) [reply package-lock.json]: @ManuGowda I think this change was not expected. I will push the change.

**Review by ManuGowda — APPROVED** [review]

File: src/application/modules/legacy/commands/reclaim.ts [hunk 3]
```diff
@@ -12,13 +12,99 @@
...
+} from '../constants';
+
+import { reclaimParamsSchema, legacyAccountSchema } from '../schemas';
+import { ReclaimParamData, LegacyStoreData, TokenIDReclaim } from '../types';
+
+const { LiskValidationError, validator } = liskValidator;
+const { getLegacyAddressFromPublicKey, getAddressFromPublicKey } = cryptography;
+
+const getLegacyAddress = (puhlicKey): Buffer =>
+	Buffer.from(getLegacyAddressFromPublicKey(puhlicKey), 'hex');
```
Comment (sameersubudhi) [inline src/application/modules/legacy/commands/reclaim.ts:39]:
```suggestion
const getLegacyAddress = (publicKey): Buffer =>
	Buffer.from(getLegacyAddressFromPublicKey(publicKey), 'hex');
```

Comment (sameersubudhi) [reply src/application/modules/legacy/commands/reclaim.ts:39]: Please fix the typos.

File: src/application/modules/legacy/types.ts [hunk 4]
```diff
@@ -16,3 +16,19 @@ export interface LegacyStoreData {
 	legacyAddress: string;
 	balance: bigint;
 }
+
+export interface ReclaimParamData {
```
Comment (sameersubudhi) [inline src/application/modules/legacy/types.ts:20]: Let us keep the naming consistent throughout.

```suggestion
export interface ReclaimParamsData {
```

**Review by sameersubudhi — APPROVED** [review]
LGTM overall 👍 
Kindly consider the minor remarks below.

**Pull request merged**